examples/simultaneous_translation/docs/evaluation.md

```diff
@@ -96,17 +103,63 @@ class Agent(object):
 
  
 ```
-Here are the implementations of agents for [text *wait-k* model](../eval/agents/simul_trans_text_agent.py) and [speech *wait-k* model](../eval/agents/simul_trans_speech_agent.py).
 
-## Quality
-The quality is measured by detokenized BLEU. So make sure that the predicted words sent to server are detokenized. An implementation is can be find [here](../eval/agent.py)
-=======
+Here is an exmaple of a customized agent. 
+First of all, a name needs to be registered. 
+Next, reload the agent functions according to the translation model. 
+The functions that need to reload are listed in `MyAgent` class as follow. 
+Finally copy the agent file to this [directory](../eval/agents) in the local fairseq repository.
+```python
+from example.simultaneous_translation.eval.agents import register_agent
+@register_agent("my_agent_name")
+class MyAgent(Agent):
+    @staticmethod
+    def add_args(parser):
+        # add customized arguments here
+
+    def __init__(self, *args, **kwargs):
+        ...
+
+    def init_states(self):
+        ...
+
+    def update_states(self, states, new_state):    
+        ...
+
+    def finish_eval(self, states, new_state):
+        ...
+    
+    def policy(self, states):
+        ...
+
+    def reset(self):
+        ...
+
+```
+
 Here are the implementations of agents for [text *wait-k* model](../eval/agents/simul_trans_text_agent.py) and [speech *wait-k* model](../eval/agents/simul_trans_speech_agent.py).
 
+Once there is implementation of the agent, to start the evaluation, 
+```
+python $fairseq_dir/examples/simultanesous_translation/eval/evaluate.py \
+    --port $port \
+    --agent-type my_agent_name \
+    --reset-server \
+    --num-threads 4 \
+    --scores \
+    --model-args ... # defined in MyAgent.add_args, such as model path, tokenizer etc.
+```
+
+It can be very slow to evaluate speech model utterence by utterence, please see [here](../scripts/start-multi-client.sh) for a faster implementation, which split the evaluation set into chunks.
```

Comment:
```suggestion
It can be very slow to evaluate the speech model utterance by utterance. See [here](../scripts/start-multi-client.sh) for a faster implementation, which split the evaluation set into chunks.
```